go/tools/gazelle/packages/package.go

```diff
@@ -15,7 +15,46 @@ limitations under the License.
 
 package packages
 
-import "strings"
+import (
+	"fmt"
+	"strings"
+)
+
+// PlatformConstraints is a map from config_setting labels (for example,
+// "@io_bazel_rules_go//go/platform:linux_amd64") to a sets of build tags
+// that are true on each platform (for example, "linux,amd64").
+type PlatformConstraints map[string]map[string]bool
+
+// ParsePlatformConstraints parses a string that contains all platforms and
+// tags on each platform. The string must contain an even number of fields,
+// separated by whitespace. Even fields (counting from 0) are platform labels.
```

Comment:
That's why I suggested just specifying the package, that way the per project specific customization is the BUILD file that lists the config settings, which has almost the same list of things in it that otherwise you have to replicate on the gazelle command line, and still have in the build file anyway. The only difference is that then you could split your config settings amongs multiple packages, which I don't think is enough of an advantage to be worth the complexity.
I feel like we ought to be trying to move towards gazelle having fewer to no command line options, not growing more.